src/librbd/ImageCtx.cc

```diff
@@ -405,13 +405,15 @@ struct C_InvalidateCache : public Context {
     return flags;
   }
 
-  int ImageCtx::snap_set(string in_snap_name)
+  int ImageCtx::snap_set(cls::rbd::SnapshotNamespace in_snap_namespace,
+			 string in_snap_name)
   {
     assert(snap_lock.is_wlocked());
-    snap_t in_snap_id = get_snap_id(in_snap_name);
+    snap_t in_snap_id = get_snap_id(in_snap_name, in_snap_namespace);
```

Comment:
Is it ok to have name and then namespace? I think this is the only place where I confused this order. I was going for name, namespace order.